cupy/_core/core.pyx

```diff
@@ -60,6 +60,20 @@ from cupy.exceptions import ComplexWarning
 NUMPY_1x = numpy.__version__ < '2'
 
 
+cdef extern from *:
+    """
+    #define _str_(s) #s
+    #define _xstr_(s) _str_(s)
+    const char* cupy_cache_key = _xstr_(CUPY_CACHE_KEY);
+    #undef _xstr_
+    #undef _str_
+    """
+    const char* cupy_cache_key  # set at build time
+
+
+CUPY_CACHE_KEY = bytes(cupy_cache_key).decode()
```

Comment:
Cython knows how to decode directly from `const char*` so it is possible to skip copying to `bytes`

```suggestion
CUPY_CACHE_KEY = cupy_cache_key.decode()
```

Reply:
Thanks, would you mind sending a follow-up PR to address this? I prefer to not re-run the CIs as explained below.

Reply:
> Thanks, would you mind sending a follow-up PR to address this? I prefer to not re-run the CIs as [explained below](https://github.com/cupy/cupy/pull/8919#discussion_r1940423577).

Reply:
Done in PR: https://github.com/cupy/cupy/pull/8923